src/openforms/authentication/signals.py

```diff
@@ -122,6 +122,11 @@ def set_auth_attribute_on_session(
     else:
         store_auth_details(instance, form_auth)
 
+    # After the authentication details have been attached to the submission, the session
+    # no longer has to keep track of this information, because the user could start another
+    # form with different / no authentication
+    del request.session[FORM_AUTH_SESSION_KEY]
```

Comment:
I looked into this, but I'm not entirely sure how I can test it locally (because it seems this is only implemented for BRP?). Judging from the explanation in the [docs](https://open-forms.readthedocs.io/en/latest/developers/extensions.html#keycloak-token-exchange-extension), I don't think it will be an issue because the `request.session[FORM_AUTH_SESSION_KEY]` is deleted after the `submission_start` signal (step 2) is sent and isn't mentioned outside of step 1.

I do however wonder if this won't cause issues for several utils in `openforms.authentication.utils` (e.g. `is_authenticated_with_plugin`), because these utils do rely on `request.session[FORM_AUTH_SESSION_KEY]`.